src/ArrayFromIterator.php

```diff
@@ -43,11 +42,7 @@
     final private function all() : PromiseSequence
     {
         return new PromiseSequence(
-            $promise = new Promise(
-                function () use (&$promise) {
-                    $promise->resolve(new ArraySequence($this->toArray()));
-                }
-            )
+            new FulfilledPromise($this->toArray())
```

Comment:
The original code stopped `toArray()` (and hence requests) from being immediately called.